Compiler/src/tfuncs.jl

```diff
@@ -1930,6 +1946,8 @@ const _tvarnames = Symbol[:_A, :_B, :_C, :_D, :_E, :_F, :_G, :_H, :_I, :_J, :_K,
     end
     return ans
 end
+@nospecs apply_type_tfunc(𝕃::AbstractLattice, headtypetype, args...) =
+    apply_type_tfunc(𝕃, pushfirst!(collect(Any, args), headtypetype))
```

Comment:
```suggestion
    apply_type_tfunc(𝕃, Any[i == 0 ? headtypetype : args[i] for i in 0:length(args)])
```
(this avoids compiling a different copy of collect for each args)

Reply:
Nice catch, thanks.